connectors/README.md

```diff
@@ -0,0 +1,30 @@
+# BigQuery and Google Cloud Storage connectors
+
+This initialization action installs specified versions of [BigQuery and Google Cloud Storage connectors](https://github.com/GoogleCloudPlatform/bigdata-interop)
+on a [Google Cloud Dataproc](https://cloud.google.com/dataproc) cluster.
+
+## Using this initialization action
+You can use this initialization action to create a new Dataproc cluster with specific versions of BigQuery and/or GCS connectors installed.
+
+```
+gcloud dataproc clusters create <CLUSTER_NAME> \
+    --initialization-actions gs://dataproc-initialization-actions/connectors/connectors.sh \
+    --metadata 'gcs-connector-version=1.7.0'
+```
+
+This script downloads specified versions of BigQuery and/or GCS connectors.
+To specify connector version, find the needed released connector versions from the
+[BigQuery and GCS connectors releases page](https://github.com/GoogleCloudPlatform/bigdata-interop/releases),
+and set one or both metadata keys `gcs-connector-version` and/or `bigquery-connector-version`.
+If version is set only for one connector (BigQuery or GCS) then only this connector will be installed, but both old BigQuery and GCS connectors will be removed.
```

Comment:
This seems more complicated as a user -- if I want to update the bigquery connector, I also need to specify a GCS connector version to keep writing to GCS?

Also, doesn't the BQ connector depend on the GCS connector?

I would vote to either mandate both are specified, or use the version map you have.

Reply:
Agree, it's not very user friendly, but it has explicit behavior and will fail-fast, if user didn't specify needed connector version during cluster creation.

In future Dataproc releases BQ connector could be (will be?) excluded from the image, so it will be weird to install BQ connector in init action if only GCS connector version is specified or require to specify both of them. But, because GCS connector will be always in Dataproc image and more widely used, it could make sense to update GCS connector even if only BQ connector version specified, but this looks illogical.

BQ connector doesn't need GCS connector to function, it has all GCS-related classes that it needs in it.

Reply:
@karth295 I have modified init action to update BigQuery connector to version that matches specified GCS connector version if BigQuery connector is installed, and do nothing if BigQuery connector not installed. PTAL.